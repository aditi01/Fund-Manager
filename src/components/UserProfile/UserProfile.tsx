import { Avatar, Drawer, Space, Table, Divider, Row, Col } from 'antd';
import { UserOutlined } from '@ant-design/icons';
import { ColumnGroupType, ColumnType } from 'antd/es/table';

interface IProps {
    isOpen: boolean,
    data: any,
    onClose: () => void
}

const UserProfile = (props: IProps) => {
    const {isOpen, data, onClose} = props

    const fetchColumns = () => {
        const columns: (ColumnGroupType<{ fundId: string; }> | ColumnType<{ fundId: string; }>)[] = [
            {
              title: 'Fund Name',
              dataIndex: 'fundName',
              key: 'fundName',
            },
            {
              title: 'Fund Type',
              dataIndex: 'fundType',
              key: 'fundType',
            },
            {
              title: 'Inception Date',
              dataIndex: 'inceptionDate',
              key: 'inceptionDate',
            },
            {
              title: 'Assets Under Management',
              dataIndex: 'assetsUnderManagement',
              key: 'assetsUnderManagement',
            },
            {
              title: '1 Year Performance',
              dataIndex: ['performance', '1year'],
              key: '1year',
            },
            {
              title: '3 Years Performance',
              dataIndex: ['performance', '3years'],
              key: '3years',
            },
            {
              title: '5 Years Performance',
              dataIndex: ['performance', '5years'],
              key: '5years',
            },
            {
              title: 'Top Holdings',
              dataIndex: 'topHoldings',
              key: 'topHoldings',
              render: (topHoldings: any[]) => (
                <ul>
                  {topHoldings.map(holding => (
                    <li key={holding.stockSymbol}>
                      {holding.stockSymbol} - {holding.percentage}
                    </li>
                  ))}
                </ul>
              ),
            },
          ];

          return columns
    }


  return (
    <>
        <Drawer
            width={window.innerWidth > 900 ? 1200 : window.innerWidth - 100}
            title={`User Data of ${data.name}`}
            placement='right'
            closable={true}
            onClose={onClose}
            open={isOpen}>

            <Space wrap size={16}>
                <Avatar size={64} icon={<UserOutlined />} />
                <b>User ID:</b><div>{data.id}</div>
            </Space>
            <Divider />
            <h4>Contact Info:</h4>

            <Row>
                <Col>Email: </Col>
                <Col>Contact No.: </Col>
            </Row>


            <Table dataSource={data.managedFunds} columns={fetchColumns()} rowKey="fundId" />

      </Drawer>
    </>
  );
};

export default UserProfile;
