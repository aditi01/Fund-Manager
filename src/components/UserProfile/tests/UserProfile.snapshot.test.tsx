// UserProfile.test.tsx
import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import UserProfile from '../UserProfile';

describe('UserProfile', () => {
  const mockData = {
    name: 'John Doe',
    id: '12345',
    email: 'john.doe@example.com',
    contactNumber: '123-456-7890',
    managedFunds: [
      {
        fundId: '1',
        fundName: 'Fund A',
        fundType: 'Equity',
        inceptionDate: '2020-01-01',
        assetsUnderManagement: '100M',
        performance: {
          '1year': '10%',
          '3years': '30%',
          '5years': '50%',
        },
        topHoldings: [
          { stockSymbol: 'AAPL', percentage: '5%' },
          { stockSymbol: 'MSFT', percentage: '3%' },
        ],
      },
      {
        fundId: '2',
        fundName: 'Fund B',
        fundType: 'Bond',
        inceptionDate: '2019-01-01',
        assetsUnderManagement: '200M',
        performance: {
          '1year': '5%',
          '3years': '15%',
          '5years': '25%',
        },
        topHoldings: [
          { stockSymbol: 'GOOGL', percentage: '6%' },
          { stockSymbol: 'TSLA', percentage: '4%' },
        ],
      },
    ],
  };

  it('renders the user profile with the given data and matches snapshot', () => {
    const { container } = render(
      <UserProfile isOpen={true} data={mockData} onClose={jest.fn()} />
    );

    expect(container).toMatchSnapshot();
  });
});
