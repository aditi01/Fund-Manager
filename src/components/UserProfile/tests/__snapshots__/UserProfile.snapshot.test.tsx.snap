// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile renders the user profile with the given data and matches snapshot 1`] = `<div />`;
